subnet/mock_registry.go

```diff
@@ -17,50 +17,55 @@ package subnet
 import (
 	"fmt"
 	"path"
-	"sort"
 	"strings"
 	"time"
 
 	etcd "github.com/coreos/flannel/Godeps/_workspace/src/github.com/coreos/etcd/client"
 	"github.com/coreos/flannel/Godeps/_workspace/src/golang.org/x/net/context"
+
+	"github.com/coreos/flannel/pkg/ip"
 )
 
-type MockSubnetRegistry struct {
-	networks map[string]*etcd.Node
-	events   chan *etcd.Response
-	index    uint64
-	ttl      time.Duration
+type netwk struct {
+	config       string
+	subnets      []Lease
+	subnetEvents chan event
 }
 
-const networkKeyPrefix = "/coreos.com/network"
+type event struct {
+	evt   Event
+	index uint64
+}
 
-func NewMockRegistry(ttlOverride time.Duration, network, config string, initialSubnets []*etcd.Node) *MockSubnetRegistry {
-	index := uint64(0)
+type MockSubnetRegistry struct {
+	networks      map[string]*netwk
+	networkEvents chan event
+	index         uint64
+}
 
-	node := &etcd.Node{Key: normalizeNetKey(network), Value: config, ModifiedIndex: 0, Nodes: make([]*etcd.Node, 0, 20)}
-	for _, n := range initialSubnets {
-		if n.ModifiedIndex > index {
-			index = n.ModifiedIndex
-		}
-		node.Nodes = append(node.Nodes, n)
-	}
+const networkKeyPrefix = "/coreos.com/network"
```

Comment:
Don't need this anymore since we're not emulating etcd.  It's only used by normalizeNetKey() at the bottom of the file, which can now be deleted since it's not called anywhere either.